lxd/instance/drivers/driver_lxc.go

```diff
@@ -1687,13 +1707,21 @@ func (d *lxc) deviceHandleMounts(mounts []deviceConfig.MountEntryItem) error {
 				}
 			}
 
-			shiftfs := false
+			idmapped := ""
 			if mount.OwnerShift == deviceConfig.MountOwnerShiftDynamic {
-				shiftfs = true
+				switch d.IdmappedStorage() {
```

Comment:
@brauner could you define a new type for the `IdmappedStorage()` function to return and then define constants for the different values it can return. This would avoid numeric comparisons. Also I was thinking that potentially it might be useful to have the constants be string type, and then update `insertMount` to accept the new type, meaning you could pass the value directly into `insertMount` and have it validate the value in there.

Reply:
Yes, it's not in final form. I just want to see the test-suite pass before I add more changes in LXC so it's low-priority for now.

Reply:
So first adapt LXC some more, more testing how this interacts with LXD and switching over more codebases, then overhaul, then actually mark it for review.

Reply:
Sounds good! Cheers